src/test/java/org/jboss/logmanager/handlers/SocketHandlerTests.java

```diff
@@ -115,71 +111,81 @@ public void testTcpPortChange() throws Exception {
     @Test
     public void testProtocolChange() throws Exception {
         Assume.assumeFalse(JDK_8219991_ERROR_MESSAGE, JDK_8219991);
-        try (SocketHandler handler = createHandler(Protocol.TCP)) {
-            try (SimpleServer server = SimpleServer.createTcpServer(port)) {
-                final ExtLogRecord record = createLogRecord("Test TCP handler");
-                handler.doPublish(record);
-                final String msg = server.timeoutPoll();
-                Assert.assertNotNull(msg);
-                Assert.assertEquals("Test TCP handler", msg);
-            }
+        // keep handler for whole test
+        SocketHandler handler;
+        try (SimpleServer server = SimpleServer.createTcpServer()) {
+            handler = createHandler(Protocol.TCP, server.getPort());
+            final ExtLogRecord record = createLogRecord("Test TCP handler");
+            handler.doPublish(record);
+            final String msg = server.timeoutPoll();
+            Assert.assertNotNull(msg);
+            Assert.assertEquals("Test TCP handler", msg);
+        }
+        // wait until the OS really release used port. https://issues.redhat.com/browse/LOGMGR-314
+        Thread.sleep(50);
 
-            // Change the protocol on the handler which should close the first connection and open a new one
-            handler.setProtocol(Protocol.SSL_TCP);
+        // Change the protocol on the handler which should close the first connection and open a new one
+        handler.setProtocol(Protocol.SSL_TCP);
 
-            try (SimpleServer server = SimpleServer.createTlsServer(port)) {
-                final ExtLogRecord record = createLogRecord("Test TLS handler");
-                handler.doPublish(record);
-                final String msg = server.timeoutPoll();
-                Assert.assertNotNull(msg);
-                Assert.assertEquals("Test TLS handler", msg);
-            }
+        try (SimpleServer server = SimpleServer.createTlsServer(handler.getPort())) {
+            final ExtLogRecord record = createLogRecord("Test TLS handler");
+            handler.doPublish(record);
+            final String msg = server.timeoutPoll();
+            Assert.assertNotNull(msg);
+            Assert.assertEquals("Test TLS handler", msg);
         }
+        // close the handler
+        handler.close();
     }
 
     @Test
     public void testTcpReconnect() throws Exception {
-        try (SocketHandler handler = createHandler(Protocol.TCP)) {
```

Comment:
Again the try/finally approach should be used here.

Reply:
same approach used.